machine-learning/_posts/2018-10-18-UsefulThingsML.md

```diff
@@ -0,0 +1,193 @@
+* Control the fraction of falsely accepted non-null hypotheses, known as the
+_false discovery rate_.
+
+
+5. **INTUITION FAILS IN HIGH DIMENSIONS**
+
+The _curse of dimensionality_ refers to either the fact that methods that work
+fine in low dimensions become intractable when the input is high-dimensional,
+or to the fact that Generalizing correctly becomes exponentially harder as
+the dimensionality (number of features) of the examples grows.
+
+Our intuitions, which come from a threedimensional world, often do not apply
+in high-dimensional ones.
+
+
+6. **THEORETICAL GUARANTEES ARE NOT WHAT THEY SEEM**the
+
+_Machine learning papers are full of theoretical guarantees_.
+
+The main role of theoretical guarantees in machine learning is not as a
+criterion for practical decisions, but as a source of understanding and
+driving force for algorithm desig
+
+Just because a learner has a theoretical justification and works in practice
+practice doesn't mean the former is the reason for the latter.
+
+
+7. **FEATURE ENGINEERING IS THE KEY**
+
+At the end of the day, some machine learning projects succeed and some fail.
+What makes the difference? Easily the most important factor is the features
+used. **Often, the raw data is not in a form that is amenable to learning, but
+you can construct features from it that are**.
+
+Machine learning is not a one-shot process of building a data set and running
+a learner, but rather an iterative process of running the learner, analyzing
+the results, modifying the data and/or the learner, and repeating.
+
+**Features that look irrelevant in isolation may be relevant in combination**.
+
+
+8. **MORE DATA BEATS A CLEVERER ALGORITHM**
+
+Pragmatically the quickest path to success is often to just get more data. As
+a rule of thumb, a dumb algorithm with lots and lots of data beats a clever
+one with modest amounts of it.
+
+Enormous mountains of data are available, but there is not enough time to
+process it, so it goes unused. Even though in principle more data means that
+more complex classifiers can be learned, **in practice simpler classifiers end
+up being used, because complex ones take too long to learn**. **Part of the
+answer is to come up with fast ways to learn complex classifiers**.
+
+**Part of the reason using cleverer algorithms has a smaller payoff than you
+might expect is that, to a first approximation, they all do the same**.
+
+**Because of the curse of dimensionality**, no existing amount of data may be
+enough. For these reasons, **clever algorithms often pay off in the end**.
+
+**In the end, the biggest bottleneck is not data or CPU cycles, but human
+cycles**.
+
+
+9. **LEARN MANY MODELS, NOT JUST ONE**
+
+The best learner varies from application to application. If instead of
+selecting the best variation found, we combine many variations, the results
+are better --often much better-- and at little extra effort for the user.
+
+Model ensembles examples: bagging, boosting,etc.

```

Comment:
Model ensembling is often better than taking a single prediction, yes, but at the cost of n-times the training and prediction time. Should that really be a suggestion ? @jodoin @ppoulin91 @CharlesAuthier @Dref360 @lemairecarl what do you think ?

Reply:
Ensembling deep models typically yields an improvement of a few %. It's mostly useful for challenges. It takes more time, yes, but it's correct to say that it requires only _little extra effort_.

Reply:
Thoughts added in 9bd2c4c.